aten/src/ATen/autocast_mode.cpp

```diff
@@ -483,11 +483,15 @@ TORCH_LIBRARY_IMPL(_, AutocastCPU, m) {
   m.fallback(torch::CppFunction::makeFallthrough());
 }
 
+
 TORCH_LIBRARY_IMPL(aten, AutocastCPU, m) {
   // lower_precision_fp cast policy
   KERNEL_CPU(ADD_NS(conv1d), "conv1d", Tensor (const Tensor &, const Tensor &, const c10::optional<Tensor>&, IntArrayRef, IntArrayRef, IntArrayRef, int64_t), lower_precision_fp)
+  KERNEL_CPU(ADD_NS(conv1d), "conv1d.padding", Tensor (const Tensor&, const Tensor&, const c10::optional<Tensor>&, IntArrayRef, c10::string_view, IntArrayRef, int64_t groups), lower_precision_fp)
   KERNEL_CPU(ADD_NS(conv2d), "conv2d", Tensor (const Tensor &, const Tensor &, const c10::optional<Tensor>&, IntArrayRef, IntArrayRef, IntArrayRef, int64_t), lower_precision_fp)
+  KERNEL_CPU(ADD_NS(conv2d), "conv2d.padding", Tensor (const Tensor&, const Tensor&, const c10::optional<Tensor>&, IntArrayRef, c10::string_view, IntArrayRef, int64_t groups), lower_precision_fp)
   KERNEL_CPU(ADD_NS(conv3d), "conv3d", Tensor (const Tensor &, const Tensor &, const c10::optional<Tensor>&, IntArrayRef, IntArrayRef, IntArrayRef, int64_t), lower_precision_fp)
+  KERNEL_CPU(ADD_NS(conv3d), "conv3d.padding", Tensor (const Tensor&, const Tensor&, const c10::optional<Tensor>&, IntArrayRef, c10::string_view, IntArrayRef, int64_t groups), lower_precision_fp)
```

Comment:
Do you have a more detailed explanation about these? Is a bit confusing.

Reply:
`aten::conv2d.padding` redispatches to `aten::convolution`, which doesn't have an autocast registered for it, `_convolution` does. Its only in the CPU kernel by invoking `_convolution` that autocast gets applied. Since the meta kernel for `aten::convolution` wont call `_convolution` without the changes no dispatching gets applied.

So I mirrored the existing pattern for `conv1d` and added it to the registration list.

Reply:
oof. I guess if we turned off autocast after we got past the autocast layer this would prevent that, but we don't really want to pay for it. :/ There may be other latent bugs like this.

Reply:
At the risk of micro-optimizing: One thing we could maybe do if we’re worried about the boxing/unboxing cost of the fallback (although we’d still have to pay for the redispatch + TLS): since we don’t actually care about the operator’s arguments, we could teach the dispatcher to not box/unbox anything if the boxed fallback that you register doesn’t take in an argument stack (indicating that it doesn’t use the arguments)